packages/core/src/components/tabs/tabs.tsx

```diff
@@ -0,0 +1,157 @@
+import { Component, Element, Event, EventEmitter, Fragment, h, Method, State } from "@stencil/core";
+import clsx from "clsx";
+
+import { Tab, TabsSwitchEvent } from "./tabs.interfaces";
+
+@Component({
+  tag: "dso-tabs",
+  styleUrl: "tabs.scss",
+  shadow: true,
+})
+export class Tabs {
+  private tabs?: Tab[];
```

Comment:
Er staat best veel logica om deze array te vullen. En dan is er nog niets gedaan voor het bijhouden daarvan (tab krijgt ander label, tab wordt toegevoegd, etc).

Wat zou je ervan vinden om dit een getter te maken die direct de DOM nodes bevraagd?

```ts
get tabs() {
  return [...this.host.querySelectorAll('dso-tabs')].map(element => ({ label: element.getAttribute('label') }));
}
```

Dan moeten we alsnog iets doen met MutationObserver om het component vers te houden, maar dan hoeven we tenminste niet meer die array en objecten actueel te houden en wellicht dat we zelfs de communicatie tussen `<dso-tabs>` en `<dso-tab>` niet meer nodig hebben.

Morgen over verder filosoferen.